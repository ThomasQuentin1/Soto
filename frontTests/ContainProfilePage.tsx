import React from "react";
import { Button } from "@material-ui/core";
import { useTranslation } from 'react-i18next';
import '../i18n'
import RightPanel from "../components/encapsulationComponents/RightPanel";
import ToggleLanguage from "../components/settings/ToggleLanguage";

const ContainProfilePage = () => {
    const [ t, i18n ] = useTranslation();

    i18n;
    return (      
    <div style={{display:"flex", flexDirection:"row", height:"100%", width:"100%", alignItems:"center"}}>
        <div style={{display:"flex", flexDirection:"column", alignItems:"center", flexGrow:8}}>
            <Button variant="contained" color="secondary">
              {t("buttonExample.label")}
            </Button>
        </div>
        <div style={{marginLeft:"30px", marginRight:"30px", fontSize:"1eh"}}>
            <p>{t('soto.description')}</p>
        </div>
        <RightPanel>
         <ToggleLanguage t={t}></ToggleLanguage>
        </RightPanel>
    </div>)
}

export default ContainProfilePage;