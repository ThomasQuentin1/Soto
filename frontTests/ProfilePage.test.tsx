import React from 'react';
import renderer from 'react-test-renderer';
import ContainProfilePage from './ContainProfilePage';

test('Creation of profile page in a component to test it', () => {
    const component = renderer.create(
        <ContainProfilePage></ContainProfilePage>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});