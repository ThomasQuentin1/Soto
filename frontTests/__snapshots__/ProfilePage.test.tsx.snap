// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creation of profile page in a component to test it 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 8,
      }
    }
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Bouton d'exemple
    </button>
  </div>
  <div
    style={
      Object {
        "fontSize": "1eh",
        "marginLeft": "30px",
        "marginRight": "30px",
      }
    }
  >
    <p>
      Hey, t'es nouveau sur Soto ? Classe tes critères selon tes convictions, mets en place tes filtres, et commence tes courses. C'est tout simple, essaye de taper le nom d'un article, sans spéficier sa marque. Soto te proposera toujours le meilleur !
    </p>
  </div>
  <div
    id="right-panel"
    style={Object {}}
  >
    <div>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Changer pour anglais
      </button>
    </div>
  </div>
</div>
`;
